doc/source/fq_nmod_sparse_mat.rst

```diff
@@ -0,0 +1,295 @@
+.. _fq-nmod-sparse-mat:
+
+**fq_nmod_sparse_mat.h** -- sparse matrixs over finite fields (word-size characteristic)
+===============================================================================
+
+Description.
+
+Types, macros and constants
+-------------------------------------------------------------------------------
+
+.. type:: fq_nmod_sparse_mat_t
+
+    Holds an array of (possibly empty) sparse vectors corresponding to rows in 
+    the matrix
+
+Memory management
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_init(fq_nmod_sparse_mat_t M, slong rows, slong cols, const fq_nmod_ctx_t ctx)
+
+    Initializes an empty sparse matrix ``M`` with given number of rows and columns
+
+.. function:: void fq_nmod_sparse_mat_clear(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Clears the entries of the matrix ``M`` and frees the space allocated for it
+
+.. function:: void fq_nmod_sparse_mat_swap(fq_nmod_sparse_mat_t M1, fq_nmod_sparse_mat_t M2, const fq_nmod_ctx_t ctx)
+
+    Swaps two matrices ``M1`` and ``M2`` (no reallocaton)
+
+
+Instantiation
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_zero(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Sets matrix ``M`` to zero (the empty sparse matrix)
+
+.. function:: void fq_nmod_sparse_mat_one(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Sets matrix ``M`` to identity matrix (based on its number of rows)
+
+.. function:: void fq_nmod_sparse_mat_set(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Makes ``N`` a (deep) copy of ``M``
+
+.. function:: void fq_nmod_sparse_mat_from_entries(fq_nmod_sparse_mat_t M, slong *rows, slong *inds, fq_nmod_struct *vals, slong nnz, const fq_nmod_ctx_t ctx)
+
+    Constructs matrix ``M`` from a given sequence of ``rows``, ``cols``, and 
+    corresponding ``vals``, all of length ``nnz``, assumes sorted by rows 
+    with no duplicate (row, col) indices
+
+.. function:: void fq_nmod_sparse_mat_append_col(fq_nmod_sparse_mat_t M, const fq_nmod_struct *v, const fq_nmod_ctx_t ctx)
+
+    Add a dense column to the right of the matrix
```

Comment:
Some clarification is needed. Is the matrix still in sparse format after this operation. Does "dense" here just refer to the format of the column?